### PR TITLE
Update memspool and panda server plugins

### PR DESCRIPTION
**Pull request by mixmasala**
This PR adds:
Updates memspool and panda services to use the new server plugin system
Makes catshadow clients send/receive cbor encoded messages
Removes length prefixed encoding where cbor already handles length prefixed decoding internally.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+FROM golang:alpine AS builder
```
Comment (david415) [inline Dockerfile:1]: why are you reverting the docker file back to an ancient version?

**Review by david415 — CHANGES_REQUESTED** [review]
merge it but first fix it so it doesn't merge that old obsolete docker file.

**Comment by mixmasala** [comment]
this PR now includes changes to panda service as well as the memspool service. the catshadow docker-based integration tests are now passing.

File: Dockerfile (hunk 1 above)
Comment (mixmasala) [reply Dockerfile]: This Dockerfile was removed on main and I guess I didn't remove it on this branch when merging main into it... Removed now.

**Comment by mixmasala** [comment]
I also removed the command padding from memspool. I'm not sure why this was padding the messages here as the sphinx payloads are also padded.

**Review by mixmasala — COMMENTED** [review]
we still need to update the reunion service as well but that makes use of a feature that lets plugins publish a set of parameters in the mixdescriptor.

we dont have a way to fetch plugin parameters with the new plugin system, though.

**Pull request merged**